### PR TITLE
Split S3 and DynamoDB storage code into two Store objects

### PR DESCRIPTION
**Pull request by bboreham**
Following the same pattern as Bigtable column key.

This is preparatory to making the schema/store config more flexible as described at #1014

### CODE REVIEW

File: pkg/chunk/aws/storage_client_s3.go [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+			cfg:       cfg.DynamoDBConfig,
+			schemaCfg: schemaCfg,
+			DynamoDB:  dynamoDB,
+		},
+		S3:         s3Client,
+		bucketName: bucketName,
+	}
+	client.queryRequestFn = client.queryRequest
+	client.batchGetItemRequestFn = client.batchGetItemRequest
+	client.batchWriteItemRequestFn = client.batchWriteItemRequest
```
Comment (tomwilkie) [inline pkg/chunk/aws/storage_client_s3.go]: Seems like we could roll them in to the struct initialisation above?

Comment (bboreham) [reply pkg/chunk/aws/storage_client_s3.go]: Yes; this is copied from what was there before: https://github.com/cortexproject/cortex/pull/1063/files#diff-ffc95905deecb289fa4b6ec35477b78aL201

**Comment by tomwilkie** [comment]
LGTM!

**Comment by bboreham** [comment]
The code for this was included in #1072

**Pull request closed without merging**